airflow/www/security.py

```diff
@@ -516,24 +515,25 @@ def _get_all_roles_with_permissions(self) -> Dict[str, Role]:
 
     def create_dag_specific_permissions(self) -> None:
         """
-        Creates 'can_read' and 'can_edit' permissions for all active and paused DAGs.
+        Creates 'can_read' and 'can_edit' permissions for all active and paused DAGs,
+        along with any `access_control` permissions provided in the DAG.
 
         :return: None.
         """
         perms = self.get_all_permissions()
-        rows = (
-            self.get_session.query(models.DagModel.dag_id)
-            .filter(or_(models.DagModel.is_active, models.DagModel.is_paused))
-            .all()
-        )
+        dagbag = DagBag(read_dags_from_db=True)
+        dagbag.collect_dags_from_db()
```

Comment:
When is this function `create_dag_specific_permissions` called?

I am just concerned if we are just moving loading of all serialized_dags to some other place

Reply:
The sync-perms cli with the `include_dags` flag set, that's it.

Previously we had this method, used during webserver startup, and the sync-perms cli with used a different approach and also supported access_control. I'm simply moving the more complete version (sync-perms cli with short circuit added) into the security manager.

Reply:
This is functionally different than what we are adding into the parsing process too, as this needs to look at all DAGs whereas parsing is done with a single DAG.

Reply:
oh cool -- as long as it is just used in the CLI it is fine .. 

Can you probably just add a comment or note in the docstring to caution a DEV on using it elsewhere

Reply:
Take a look now. That work?